docs/content/docs/operators/feature/interaction.md

```diff
@@ -0,0 +1,169 @@
+---
+title: "Interaction"
+weight: 1
+type: docs
+aliases:
+- /operators/feature/interaction.html
+---
+
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+## Interaction
+
+Interaction takes vector or numerical columns, and generates a single vector column that contains
+the product of all combinations of one value from each input column.
+
+For example, when the input feature values are Double(2) and Vector(3, 4), the output would be 
+Vector(6, 8). When the input feature values are Vector(1, 2) and Vector(3, 4), the output would
+be Vector(3, 4, 6, 8). If you change the position of these two input Vectors, the output would 
+be Vector(3, 6, 4, 8).
+
+### Input Columns
+
+| Param name | Type   | Default | Description               |
+|:-----------|:-------|:--------|:--------------------------|
+| inputCols  | Vector | `null`  | Columns to be interacted. |
+
+### Output Columns
+
+| Param name | Type   | Default    | Description    |
+|:-----------|:-------|:-----------|:---------------|
+| outputCol  | Vector | `"output"` | Result vector. |
```

Comment:
Should we make the description for the output vector consistent? e.g., "Interacted vector"?

Same for other output vectors.